sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/utils.scala

```diff
@@ -145,49 +147,46 @@ object Utils {
 
     // 2. Create an Aggregate Operator for final aggregations.
     val finalAggregateExpressions = aggregateExpressions.map(_.copy(mode = Final))
-    val finalAggregateAttributes =
-      finalAggregateExpressions.map {
-        expr => aggregateFunctionMap(expr.aggregateFunction, expr.isDistinct)._2
-      }
+    // The attributes of the final aggregation buffer, which is presented as input to the result
+    // projection:
+    val finalAggregateAttributes = finalAggregateExpressions.map {
+      expr => aggregateFunctionMap(expr.aggregateFunction, expr.isDistinct)._2
```

Comment:
After unifying the rewriting of the result expressions, I think that we no longer access the first field of this map's result. Therefore, I'm going to try to clean this up by changing the type of the map, eliminating the `._2`, and re-naming it to something more descriptive.